apps/src/utils.js

```diff
@@ -431,3 +433,32 @@ exports.ellipsify = function (inputText, maxLength) {
   }
   return inputText || '';
 };
+
+/**
+ * Returns deep merge of two objects, concatenating rather than overwriting
+ * array properites. Does not mutate either object.
+ *
+ * Note: new properties in overrides are always added to end, not in-order.
+ *
+ * Note: may become default behavior of mergeDeep in future immutable versions.
+ *   @see https://github.com/facebook/immutable-js/issues/406
+ *
+ * @param {Object} baseObject
+ * @param {Object} overrides
+ * @returns {Object} original object (now modified in-place)
+ */
+exports.deepMergeConcatArrays = (baseObject, overrides) => {
+  function deepConcatMerger(a, b) {
+    const isList = Immutable.List.isList;
+    if (isList(a) && isList(b)) {
+      return a.concat(b);
+    }
+    if (a && a.mergeWith) {
+      return a.mergeWith(deepConcatMerger, b);
+    }
+    return b;
+  }
+
+  var baseImmutable = Immutable.fromJS(baseObject);
+  return baseImmutable.mergeWith(deepConcatMerger, overrides).toJS();
```

Comment:
Added TODO note to swap out `[Immutable].mergeWith` with `_.mergeWith` once lodash upgraded to 4.x. Slightly prefer having the mutability behavior of this util specified, and will be an easy swap-in as lodash becomes available (confirmed _.mergeWith technique passes same unit tests).